doc/api/process.md

```diff
@@ -337,7 +337,8 @@ changes:
   rejection or from an synchronous error. Can either be `'uncaughtException'` or
   `'unhandledRejection'`. The latter is only used in conjunction with the
   [`--unhandled-rejections`][] flag set to `strict` or `throw` and
-  an unhandled rejection.
+  an unhandled rejection, or when a rejection happens during the ES module
+  static loading phase.
```

Comment:
can we just simplify this to `when in a Promise based async context`? Since purely with CJS and having no ESM this also happens in some cases.

Reply:
Do you have a purely CJS example when this happen when `--unhandled-rejections` flag is NOT set to `strict` or `throw`?

Reply:
@aduh95 I do not for that case, but it does happen outside of ES module static loading phase.

Reply:
> it does happen outside of ES module static loading phase

Any chance we can list those edge cases as well? I don't think `when in a Promise based async context` does a great job at explaining when this will come up – my take away from https://github.com/nodejs/node/issues/41328 is that it's not obvious to everyone that ESM loading is `Promise` based, so imho it's worth explicitly list it here rather than expect readers to know what is a `Promise` based context.

Reply:
> Do you have a purely CJS example when this happen when `--unhandled-rejections` flag is NOT set to `strict` or `throw`?

Didn’t the default for `--unhandled-rejections` change in 17? So it doesn’t need to be set anymore to get the error.

Reply:
It changed in v15.0.0. But that's beside the point I think, this section of the docs is currently wrong, and if this PR doesn't do a good enough job to document when users should expect `unhandledRejection` rather than `uncaughtException`, I'm open to suggestions.

Reply:
@aduh95 roughly anything that creates a Promise job causes that, so IDK where to start. If an uncaughtException would occur inside of an async function / inside of a Promise constructor/handler callbacks / inside ESM are all the real times that this happens. So uncaughtException isn't really about the source file / throwing at the top level, but it is about how it was invoked, always. Listing async functions and Promise methods might be enough?

Reply:
async functions, promise then/catch/finally, new Promise executor functions, and soon, Array.fromAsync

Reply:
> anything that creates a Promise job causes that, so IDK where to start

@bmeck I think that is true when `--unhandled-rejections` is set to `strict` or `throw` (which is default), and we're in agreement on this part. I'll try to rephrase the sentence so this part is clearer.

What is more tricky is that a rejection in the ES loader will always trigger an `'uncaughtException'` event with `origin` `'unhandledRejection'`, even if `--unhandled-rejections` is set to `none` (or `warn` or `warn-with-error-code`). At least to me, this part is surprising, and in contradiction with what the docs are currently saying, so I think it's worth documenting it. I _believe_ that the ES module loader is the only piece of node that behaves like this, but if you know of another example, I can include it.

Reply:
I can't think of another off the top of my head. I think we've done enough workshopping though that whatever is clarified seems good at this point.